src/Conversion/KrnlToAffine/ConvertKrnlToAffine.cpp

```diff
@@ -778,6 +782,23 @@ static LogicalResult interpretOperation(Operation *op, OpBuilder &builder,
       parallelLoop.getRegion().takeBody(loopToParallel.getRegion());
       Operation *yieldOp = &parallelLoop.getBody()->back();
       yieldOp->setOperands(reducedValues);
+      if (needParallelClause) {
+        // Use clause only for the first one (expected the outermost one).
+        // Ideally, we would generate here a single, multi-dimensional
+        // AffineParallelOp, and we would not need to reset the flag.
+        needParallelClause = false;
+        // Currently approach: insert after yield and then move before it.
+        PatternRewriter::InsertionGuard insertGuard(builder);
+        builder.setInsertionPointAfter(yieldOp);
```

Comment:
Doesn't `setInsertionPoint(yieldOp)` work for inserting just before `yieldOp`?

Reply:
For some reasons, if I don't have the "moveBefore", it get's me this error
```
flt_orig_model.mlir:18:3: error: operand #0 does not dominate this use
  krnl.iterate(%loop_block) with (%0 -> %arg1 = 0 to 16384){
  ^
flt_orig_model.mlir:18:3: note: see current operation: "krnl.parallel_clause"(%arg1, %0) {proc_bind = "spread"} : (index, i32) -> ()
flt_orig_model.mlir:18:3: note: operand defined as a block argument (block #0 in a child region)
```
Strangely, with the `moveBefore(yieldOp)`, I get the same result with the `setInsertionPointAfter` or `setInsertionPoint`.
There is something fragile about the lowering of Krnl to Affine with respect to "movable". 

Since it works as is, I prefer to leave it that way.

Reply:
This conversion pass traverses the IR by ourselves. We manipulate graph directly. That might be the reason why it is fragile.

Reply:
OK, then it's fine with the current way.